include/lemlib/PID.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "units/units.hpp"
+
+namespace lemlib {
+/**
+ * @brief Struct to hold PID gains.
+ *
+ * @param kP proportional gain
+ * @param kI integral gain
+ * @param kD derivative gain
+ */
+struct Gains {
+        double kP = 0;
+        double kI = 0;
+        double kD = 0;
+};
+
+class PID {
+    public:
+        /**
+         * @brief Constructs a new PID controller
+         *
+         * @param kP proportional gain
+         * @param kI integral gain
+         * @param kD derivative gain
+         * @param windupRange range at which integral is reset
+         * @param signFlipReset whether to reset integral when error changes sign
+         */
+        PID(double kP, double kI, double kD, double windupRange = 0, bool signFlipReset = false);
+        /**
+         * @brief Constructs a new PID controller
+         *
+         * @param gains the gains to use
+         * @param windupRange range at which integral is reset
+         * @param signFlipReset whether to reset integral when error changes sign
+         */
+        PID(const Gains& gains, double windupRange = 0, bool signFlipReset = false);
+        /**
+         * @brief Get the current gains
+         *
+         * @return Gains the current gains
+         */
+        Gains getGains();
+        /**
+         * @brief Set the new gains
+         *
+         * @param gains the new gains
+         */
+        void setGains(Gains gains);
+        /**
+         * @brief Updates the PID controller using a given error, and outputs the next control signal.
+         *
+         * @param error the error from the setpoint. Error is calculated as setpoint - current
+         * @return double the control signal (output)
+         */
+        double update(double error, Time dt);
```

Comment:
There's no reason this function should take the time delta. That can be calculated in the implementation